WORKSPACE

```diff
@@ -16,6 +16,32 @@ load("@com_google_protobuf//:protobuf_deps.bzl", "protobuf_deps")
 
 protobuf_deps()
 
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+http_archive(
+    name = "rules_jvm_external",
+    sha256 = "62133c125bf4109dfd9d2af64830208356ce4ef8b165a6ef15bbff7460b35c3a",
+    strip_prefix = "rules_jvm_external-3.0",
+    url = "https://github.com/bazelbuild/rules_jvm_external/archive/3.0.zip",
+)
+
+load("@rules_jvm_external//:defs.bzl", "maven_install")
+load("@io_grpc_grpc_java//:repositories.bzl", "IO_GRPC_GRPC_JAVA_ARTIFACTS")
+load("@io_grpc_grpc_java//:repositories.bzl", "IO_GRPC_GRPC_JAVA_OVERRIDE_TARGETS")
+
+maven_install(
```

Comment:
Most probably this will break generation of gradle build scripts for standalone clients (also part of self-serivce).

Reply:
That's surprising. I thought the gradle build and the maven build were independent.

All tests pass. What test are we missing?

Reply:
The tests test gax build itself, but they do not test what will happen when it gets integrated with googleapis/googleapis repository. Since that repo pulls million different dependencies (including gax and its transitive dependencies) things get complicated.

To fully test it it must be integrated with `googleapis/googleapis`, then try building any of the java targets from there (like `bazel build //google/cloud/language/v1:google-cloud-language-v1-java`, then untar the created `bazel-bin/google/cloud/language/v1/google-cloud-language-v1-java.tar.gz` archive, go there and run `./gradlew clean test`).

To integrate the googleapis/googleapis with your changes you can replace `https://github.com/googleapis/googleapis/blob/master/WORKSPACE#L161` with the following:

```
local_repository(
   name = "com_google_api_gax_java, 
   path = "/absolute/path/to/your/changes/locally
)
```

Some of the values from the dependencies file are used to generate gradle build files: https://github.com/googleapis/gapic-generator/blob/master/rules_gapic/java/resources/gradle/proto.gradle.tmpl#L21

Also, if we depend on `maven_install`, I'm not sure how that will resolve potential conflicts between pulled libraries. Now, we control all the pulling manually and they are under "_maybe", so if one dependency is pulled with the same name, and then another dependency with the same name is encountered, the first one "wins" and the second one gets ignored. it is not very convenient, but at least give  us full control on what actually gets used during the build.

Reply:
Can you please elaborate on why it is not optional? Both "styles" of importing maven dependencies are supposed to be supported by bazel regardless of what chooses to do.

Reply:
see @ejona's comment at https://github.com/googleapis/gax-java/pull/1263#discussion_r566997807

With the latest gRPC we can no longer use the old style of imports.

Reply:
You can still use `jvm_maven_import_external()` at this point. But we don't help you. And it is a giant pain. I don't expect anyone to really be doing it themselves because it is so painful. But we are still supporting the approach for the while since there are other dependency management systems and they use `jvm_maven_import_external()` or similar internally. I was going to look at the ecosystem in a year or so and try to determine what direction things were going in.